The libdvdread version of nav_types.h now differs from the actual disk 
layout, which is problematic for libdvdread. We provide a patched 
version of libdvdread's nav_types.h for internal use.

--- libdvdread/dvdread/nav_types.h	2003-03-06 23:38:10.000000000 +0100
+++ libdvdnav/src/nav_types.h	2003-03-06 23:20:24.000000000 +0100
@@ -144,10 +144,6 @@
 
 /** 
  * Button Information
- *
- * NOTE: I've had to change the structure from the disk layout to get
- * the packing to work with Sun's Forte C compiler.
- * The 4 and 7 bytes are 'rotated' was: ABC DEF GHIJ  is: ABCG DEFH IJ
  */
 typedef struct {
 #ifdef WORDS_BIGENDIAN
@@ -156,14 +152,13 @@
   unsigned int zero1            : 2;  /**< reserved */
   unsigned int x_end            : 10; /**< x end offset within the overlay */
 
-  unsigned int zero3            : 2;  /**< reserved */
-  unsigned int up               : 6;  /**< button index when pressing up */
-
   unsigned int auto_action_mode : 2;  /**< 0: no, 1: activated if selected */
   unsigned int y_start          : 10; /**< y start offset within the overlay */
   unsigned int zero2            : 2;  /**< reserved */
   unsigned int y_end            : 10; /**< y end offset within the overlay */
 
+  unsigned int zero3            : 2;  /**< reserved */
+  unsigned int up               : 6;  /**< button index when pressing up */
   unsigned int zero4            : 2;  /**< reserved */
   unsigned int down             : 6;  /**< button index when pressing down */
   unsigned int zero5            : 2;  /**< reserved */
@@ -176,14 +171,13 @@
   unsigned int x_start          : 10;
   unsigned int btn_coln         : 2;
 
-  unsigned int up               : 6;
-  unsigned int zero3            : 2;
-
   unsigned int y_end            : 10;
   unsigned int zero2            : 2;
   unsigned int y_start          : 10;
   unsigned int auto_action_mode : 2;
 
+  unsigned int up               : 6;
+  unsigned int zero3            : 2;
   unsigned int down             : 6;
   unsigned int zero4            : 2;
   unsigned int left             : 6;
